### PR TITLE
[DOC] Adding documentation for experiments

### PR DESCRIPTION
**Pull request by sangm**
As I'm picking up more of Ember CLI work, I realized we are missing
documentation on how experiments work.

I synced with @rwjblue offline, and I know the information here isn't fully
clear. Would like some help from other contributors.

### CODE REVIEW

File: docs/experiments.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+} else {
+  ...
+}
+```
+
+If the experiment is not ready to progress to beta, we remove the experiment and
+feature is disabled.
+
+## Supported
+Once an experiment has through the different betas, and we can confidently say
```
Comment (twokul) [inline docs/experiments.md]: are you missing a word here or am I misreading? once an experiment has through what?

File: docs/experiments.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+if (experiments.MODULE_UNIFICATION) {
+  ...
+} else {
+  ...
+}
+```
+
+## Unsupported
+The Ember CLI core team will evaluate each experiment before betas get released.
+If the experiment is not ready, experiment is deleted from the codebase.
```
Comment (rwjblue) [inline docs/experiments.md]: I would rephrase this to:

> If the experiment is not ready, the entry for the experiment is deleted from `lib/experiments/index.js`.

File: docs/experiments.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+## Supported
+Once an experiment has gone through the different stages of beta, and we can
+confidently say a specific feature from an experiment will be supported, we
+delete the exerpiemnt from the codebase.
```
Comment (rwjblue) [inline docs/experiments.md]: typo here

Comment (rwjblue) [reply docs/experiments.md]: I would tweak the verbiage to:

> confidently say a specific feature from an experiment will be supported, we delete the entry in `lib/experiments/index.js` and remove the experiment guards (e.g. `if (experiments.FOO_BAR) {`) from the codebase.

**Comment by rwjblue** [comment]
Thanks @sangm!
